armadillo/src/test/java/org/molgenis/armadillo/storage/CharacterSeparatedFileTest.java

```diff
@@ -0,0 +1,113 @@
+package org.molgenis.armadillo.storage;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+
+import com.opencsv.exceptions.CsvValidationException;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.molgenis.armadillo.exceptions.FileProcessingException;
+import org.springframework.web.multipart.MultipartFile;
+
+class CharacterSeparatedFileTest {
+
+  MultipartFile mockFile;
+
+  @BeforeEach
+  void setup() {
+    mockFile = mock(MultipartFile.class);
+  }
+
+  @Test
+  void testConstructorWithCommaSeparator() throws IOException, CsvValidationException {
+    String csvData = "name,age\nJohn,30\nJane,25\n";
+    Mockito.when(mockFile.getInputStream())
+        .thenReturn(new ByteArrayInputStream(csvData.getBytes()));
+    Mockito.when(mockFile.getOriginalFilename()).thenReturn("test.csv");
+
+    CharacterSeparatedFile csf = new CharacterSeparatedFile(mockFile);
+
+    assertNotNull(csf);
+    assertEquals(',', csf.separator);
```

Comment:
probably should be getters and the properties private? Or are these properties immutible?